fuel-core/src/executor.rs

```diff
@@ -115,6 +121,23 @@ struct ExecutionData {
     tx_status: Vec<(Bytes32, TransactionStatus)>,
 }
 
+pub trait TxnStatusSender: Send + Sync {
```

Comment:
Hmm, I think it is better to move notification logic to the upper level because execution should only execute the block and it doesn't know is block committed fully or not. The `CM`(consensus module, currently PoA) knows how to process the execution result, so it fits better there.

For example, I moved logic about the skipped transactions to [`PoACoordinator`](https://github.com/FuelLabs/fuel-core/pull/751/files#diff-57b62c1d2fc219782560c965fd5c648187323526bc52c95cfcea5f15c887f263). I added `ExecutionResult` struct that contains some meta information gathered during block execution. We can add transaction statutes there too.

After reviewing your PR, it seems better for you and me to move logic regarding transactions directly to `TxPool`. So execution can return `ExecutionResult` with block, skipped transactions, and transactions statuses to `CM`, and `CM` will commit the block and propagate skipped transactions and statuses to `TxPool`. `TxPool` will clean up all transactions and update statuses.

I'm not sure about the idea of moving to `TxPool`(I'm waiting for your feedback), but I'm sure that we need to move it upper(`CM` for example)=)

Reply:
@Voxelot Thoughts on this ☝️

Reply:
I don't fully understand this because once the block is committed to the db what else is done to make it "fully committed"?

Reply:
This makes sense, as we already send an "import block event" to the txpool, which would allow us to consolidate sending tx status updates to the txpool.

Reply:
Ok I have changed this to the tx pool. It was actually a bit of work but should be all good now. I refactored the braodcast channel a bit because sending extra information to places it wasn't used was getting hard to handle.

Reply:
> I don't fully understand this because once the block is committed to the db what else is done to make it "fully committed"?

Based on our architecture proposal executor should only execute the block. It is not expected that the executor will commit the block. We do it now because before, we didn't though about it, but we will re-work it in the future.

Also, it seems it is easy to fix. When we produce a block, we can use a temporary database(`DatabaseTransaction`) and apply all changes to it. After execution, we can return this `DatabaseTransaction` inside of the execution result, and `CM` or `BI` will call the `ExecutionResult::commit` method.